server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -650,6 +661,23 @@ public ClusterState execute(ClusterState currentState) {
                         return allocationService.reroute(updatedState, "restored snapshot [" + snapshot + "]");
                     }
 
+                    private void validateReplicationTypeRestoreSettings(String snapshotReplicationType,
```

Comment:
nitpicks - 

Lets pass a ReplicationType here and parse in the caller and use `ReplicationType. DOCUMENT.equals(replicationType)`

is `snapshotIndexMetadata` the metadata inside the snapshot? Passing that as `restoreReplicationType` is a bit confusing.  How about `snapshottedMetadata` and `updatedMetadata` ?

Reply:
`snapshotIndexMetadata` contains metadata of snapshot and whatever setting user is trying to override. So basically its a metadata which will be applied for resulting restored index.

Reply:
do you mean like this?
```
private void validateReplicationTypeRestoreSettings(IndexMetadata snapshottedMetadata, IndexMetadata updatedMetadata)
```